docs/guides/compile-contracts.md

```diff
@@ -9,19 +9,91 @@ Compiled 1 contract successfully
 
 The compiled artifacts will be saved in the `artifacts/` directory by default, or whatever your configured artifacts path is. Look at the [paths configuration section](../config) to learn how to change it. This directory will be created if it doesn't exist.
 
-After the initial compilation, if the contract code hasn't changed then Hardhat will skip compilation when running the `compile` task:
+After the initial compilation, Hardhat will try to do the least amount of work possible the next time you compile. For example, if you didn't change any file since the last compilation, nothing will be compiled. If you only modified one file, only that file and others affected by it will be recompiled.
+
 ```
 $ npx hardhat compile
-All contracts have already been compiled, skipping compilation.
+Nothing to compile
 ```
 
 To force a compilation you can use the `--force` argument, or run `npx hardhat clean` to clear the caches and delete the artifacts.
 
+## Configuring the compiler
+
+If you need to customize the `solc` compiler options, then you can do so through the `solidity` config field in your `hardhat.config.js`. The simplest way to use this field is the shorthand for setting the compiler version, which we recommend always doing:
```

Comment:
> which we recommend always doing

It's just me or this can be misleading? It sounds like we recommend to always configure the compiler this way, instead of "we recommend always setting some version of the compiler".